src/org/thoughtcrime/securesms/ConversationListFragment.java

```diff
@@ -380,34 +380,34 @@ private void handleCreateConversation(int chatId) {
     ((ConversationSelectedListener)getActivity()).onCreateConversation(chatId);
   }
 
+  private final Object loadChatlistLock = new Object();
   private boolean inLoadChatlist;
   private boolean needsAnotherLoad;
   private void loadChatlistAsync() {
-    if (inLoadChatlist) {
+    synchronized (loadChatlistLock) {
```

Comment:
If the chatlist is loading at this moment in another thread, we will have to wait for it to finish before entering this `synchronized` instead of simply setting `needsAnotherLoad = true` and returning.

Reply:
the lock is not acquired while `loadChatlist()` is executing, only in some vars assignments and checking and that should be fast, or maybe I didn't understand something :sweat_smile:

Reply:
Yes, lock is not acquired, so your thread is waiting at this line. Instead, it should set `needsAnotherLoad = true` and return immediately.

Reply:
I meant that lock can be acquired here while `loadChatlist()` is executing

Reply:
I overlooked that `synchronized` does not span over the `loadChatlist`